test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -880,57 +895,79 @@ define([
+                it(`should not send a ${msgType} message, one job, invalid`, () => {
+                    const comm = new JobCommChannel(),
+                        msgData = { [TEST_JOB_ID]: value },
+                        msg = makeCommMsg(msgType, msgData);
+
                     return comm.initCommChannel().then(() => {
                         spyOn(comm, 'reportCommMessageError');
                         spyOn(console, 'error');
                         comm.handleCommMessages(msg);
                         expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
-                            [{ msgType, msgData: value }],
+                            [{ msgType, msgData }],
                         ]);
                     });
                 });
             });
 
-            if (type !== 'Logs') {
-                it(`should not send a ${msgType} message with invalid data`, () => {
-                    const allMsgData = {};
-                    values.forEach((val, ix) => {
-                        allMsgData[`test_job_${ix}`] = val;
-                    });
-                    const msg = makeCommMsg(msgType, allMsgData),
-                        comm = new JobCommChannel();
-                    return comm.initCommChannel().then(() => {
-                        spyOn(comm, 'reportCommMessageError');
-                        spyOn(console, 'error');
-                        comm.handleCommMessages(msg);
-                        expect(comm.reportCommMessageError.calls.allArgs()).toEqual(
-                            values.map((msgData) => {
-                                return [{ msgType, msgData }];
-                            })
-                        );
-                    });
+            // message where all data is invalid
+            it(`should not send a ${msgType} message, many jobs, invalid`, () => {
+                const msg = makeCommMsg(msgType, expected.invalid),
+                    comm = new JobCommChannel();
+                return comm.initCommChannel().then(() => {
+                    spyOn(comm, 'reportCommMessageError');
+                    spyOn(console, 'error');
+                    comm.handleCommMessages(msg);
+                    expect(comm.reportCommMessageError.calls.allArgs()).toEqual([
+                        [{ msgType, msgData: expected.invalid }],
+                    ]);
                 });
-            }
-        });
+            });
 
-        it('Should handle unknown messages with console warnings', () => {
-            const comm = new JobCommChannel(),
-                msg = makeCommMsg('unknown_weird_msg', {});
-            return comm.initCommChannel().then(() => {
-                spyOn(console, 'warn');
-                comm.handleCommMessages(msg);
-                expect(console.warn).toHaveBeenCalled();
+            // message with a mix of valid and invalid data
+            it(`should separate out valid and invalid ${type} data`, () => {
```

Comment:
make sure that if a message comes in with a mix of good and bad data, the bad data is picked out and reported, and the good data gets sent off on the bus.